build_mt.py

```diff
@@ -404,61 +391,33 @@ def main(sim_name, z_start, z_stop, merger_parent, catalog_parent, resume=False,
+        # reading the "high" slab at the leading edge, discarding the trailing "low" slab
+        # and moving the mid to low. But first we need to read all three to prime the queue
+        mt_prev = {}  # indexed by slab num
+        mt_prev[(first_ss-1)%n_chunks] = get_mt_info(fns_prev[(first_ss-1)%n_chunks], fields=fields_mt, minified=minified)
+        mt_prev[first_ss] = get_mt_info(fns_prev[first_ss], fields=fields_mt, minified=minified)
 
         # for each chunk
-        for k in range(n_chunks):
-
-            # starting and finishing superslab chunks; 
-            inds_fn_this = [k]
-            inds_fn_prev = np.array([(k-1),k,(k+1)],dtype=int) % n_chunks
+        for k in range(first_ss,n_chunks):
+            # starting and finishing superslab chunks
+            klow = (k-1)%n_chunks
+            khigh = (k+1)%n_chunks
+            
+            # Slide down by one
+            if (klow-1)%n_chunks in mt_prev:
+                del mt_prev[(klow-1)%n_chunks]
+            mt_prev[khigh] = get_mt_info(fns_prev[khigh], fields=fields_mt, minified=minified)
 
-            print("chunks loaded in this and previous redshifts = ",inds_fn_this, inds_fn_prev)
+            print(f"Loaded chunk {k} in this redshift, and {tuple(mt_prev)} in previous")
             # get merger tree data for this snapshot and for the previous one
-            if "Progenitors" in fields_mt:
-                (
-                    Merger_this,
-                    Progs_this,
-                    N_halos_slabs_this,
-                    slabs_this,
-                ) = get_mt_info(
-                    fns_this,
-                    fields=fields_mt,
-                    minified=minified,
-                    inds_fn=inds_fn_this
-                )
-                (
-                    Merger_prev,
-                    Progs_prev,
-                    N_halos_slabs_prev,
-                    slabs_prev
-                ) = get_mt_info(
-                    fns_prev,
-                    fields=fields_mt,
-                    minified=minified,
-                    inds_fn=inds_fn_prev,
-                )
-            else:
-                (
-                    Merger_this,
-                    N_halos_slabs_this,
-                    slabs_this,
-                ) = get_mt_info(
-                    fns_this,
-                    fields=fields_mt,
-                    minified=minified,
-                    inds_fn=inds_fn_this,
-                )
-                (
-                    Merger_prev,
-                    N_halos_slabs_prev,
-                    slabs_prev,
-                ) = get_mt_info(
-                    fns_prev,
-                    fields=fields_mt,
-                    minified=minified,
-                    inds_fn=inds_fn_prev,
-                )
-
+            mt_data_this, halos_per_slab_this = get_mt_info(fns_this[k], fields=fields_mt, minified=minified)
+            
+            # ======== LHG: haven't edited below here
```

Comment:
The main change needed to use the rolling algorithm is that one has three separate arrays in which to do the progenitor lookup.  But you don't need to check all three, because the progenitor index tells you which one to look in.  I would definitely keep the arrays separate in memory rather than try to merge them.